test/services/weather_request_service_test.rb

```diff
@@ -0,0 +1,58 @@
+    @json_from_bad_location_id = '{"query":{"count":1,"created":"2018-10-27T06:13:26Z","lang":"en-AU","results":{"channel":{"units":{"distance":"mi","pressure":"in","speed":"mph","temperature":"F"}}}}}'
+    @json_where_location_not_found = '{"error":{"lang":"en-US","description":"Invalid identfier asdfghjkl. me AND me.ip are the only supported identifier in this context"}}'
+    @invalid_json = '{ "really bad" { "json"'
+  end
+
+  test "valid request" do
+    Net::HTTP.expects(:get).returns(@json_valid)
+    assert_difference 'WeatherRequest.count', 1 do
+      @weather_request = WeatherRequestService.new(@location).run
+    end
+    assert_equal @weather_request.returned_json, @json_valid
+    assert_equal @weather_request.status,        WeatherRequest::SUCCESS
```

Comment:
Do you think it is worth adding a predicate, and then just using `assert`?

Reply:
I think after I wrote these tests, I had added a `successful?` method to WeatherRequest.  I'm also not used to minitest (I use rspec at work), and I didn't realize there was an assert_predicate method.  I guess I would rewrite as...

`assert_predicate @weather_request, :successful?`
 
or just...

`assert @weather_request.successful?`

Reply:
Yes, agreed, but the simple assert open is better with a message that helps you when it fails, e.g. something like,
```ruby
  assert @weather_request.successful?, "simple weather request should be successful"
```
With Minitest, it would say something like, "expected true but got false", which doesn't help when you've mucked up something.

Reply:
ok, thanks.  I'm not used to minitest.